ros_buildfarm/templates/__init__.py

```diff
@@ -24,8 +24,51 @@
 import warnings
 from xml.sax.saxutils import escape
 
-from em import BANGPATH_OPT
-from em import Hook
+try:
+    # Import bangpath option name from EmPy v3
+    from em import BANGPATH_OPT
+    BANGPATH_VALUE = False
+except ImportError:
+    # EmPy 4 does not define an import-able constant for the updated bangpath
+    # option and inverts its meaning.
+    BANGPATH_OPT = 'ignoreBangpaths'
+    BANGPATH_VALUE = True
+
+try:
+    # EmPy v3
+    from em import Hook
+except ImportError:
+    # EmPy v4
+    from emlib import Hook as _Hook
+
+    class Hook(_Hook):
+        """Hook implementation with compatibility back to EmPy v3."""
+
+        def beforeFileLines(self, file, locals, *args, **kwargs):
+            return self.beforeFile(None, file, locals)
+
+        def afterFileLines(self):
+            return self.afterFile()
+
+        def beforeFileChunks(self, file, bufferSize, locals, *args, **kwargs):
+            return self.beforeFile(None, file, locals)
+
+        def afterFileChunks(self):
+            return self.afterFile()
+
+        def beforeFileFull(self, file, locals, *args, **kwargs):
+            return self.beforeFile(None, file, locals)
+
+        def afterFileFull(self):
+            return self.afterFile()
+
+        def beforeFile(self, name, file, locals):
+            pass
+
+        def afterFile(self):
+            pass
```

Comment:
Do we need to emit any kind of warning here or do we expect all downstream consumers to call the other methods?